src/serializers/decoders/current/v10/decode_graph.c

```diff
@@ -0,0 +1,220 @@
+/*
+* Copyright 2018-2020 Redis Labs Ltd. and Contributors
+*
+* This file is available under the Redis Labs Source Available License Agreement
+*/
+
+#include "decode_v10.h"
+
+static GraphContext *_GetOrCreateGraphContext
+(
+	char *graph_name
+) {
+	GraphContext *gc = GraphContext_GetRegisteredGraphContext(graph_name);
+	if(!gc) {
+		// New graph is being decoded. Inform the module and create new graph context.
+		gc = GraphContext_New(graph_name, GRAPH_DEFAULT_NODE_CAP, GRAPH_DEFAULT_EDGE_CAP);
+		// While loading the graph, minimize matrix realloc and synchronization calls.
+		Graph_SetMatrixPolicy(gc->g, SYNC_POLICY_RESIZE);
+	}
+	// Free the name string, as it either not in used or copied.
+	RedisModule_Free(graph_name);
+
+	// Set the GraphCtx in thread-local storage.
+	QueryCtx_SetGraphCtx(gc);
+
+	return gc;
+}
+
+/* The first initialization of the graph data structure guarantees that there will be no further re-allocation
+ * of data blocks and matrices since they are all in the appropriate size. */
+static void _InitGraphDataStructure(Graph *g, uint64_t node_count, uint64_t edge_count,
+									uint64_t label_count,  uint64_t relation_count) {
+	Graph_AllocateNodes(g, node_count);
+	Graph_AllocateEdges(g, edge_count);
+	for(uint64_t i = 0; i < label_count; i++) Graph_AddLabel(g);
+	for(uint64_t i = 0; i < relation_count; i++) Graph_AddRelationType(g);
+	// flush all matrices, guarantee matrix dimensions matches graph's nodes count
+	Graph_ApplyAllPending(g, true);
+}
+
+static GraphContext *_DecodeHeader(RedisModuleIO *rdb) {
+	/* Header format:
```

Comment:
Please update this comment with the changes we've make to the format, make sure the encoder has a similar update to its comment.